src/movies.js

```diff
@@ -1,17 +1,44 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+// console.log('movies: ', movies);
+
+const movies = require("./data");
 
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(element) {
+  const allDirectors = [];
+  movies.map( function(element, index) {
+      return true
+  })
+  return allDirectors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+  let count = 0
+  const spielbergMovies = movies.filter(function(element) {
+    if(movies.includes('Steven Spielberg') && movies.includes('Drama')) {
+      count += 1
+    } 
+  })
```

Comment:
in iteration2 we need to filter by two things: director and genre. 

- we checkif it's drama  by accessing genre array - > as you done
- we check if director is spielberg by extracting movie director and checking if strings are matching

```
if (element.director === 'Steven Spielberg' && element.genre.includes('Drama'))
```

and if we pass the condition -> we `return true `, no need for counter as this method already does that for us :) 

Also if you have notice in my example, I am actually accessing properties of element not the movies. `movies` is your full array, while `element` is the name you gave -> for every element you are checking